mmseg/datasets/custom.py

```diff
@@ -227,17 +224,24 @@ def prepare_test_img(self, idx):
     def format_results(self, results, **kwargs):
         """Place holder to format result to dataset specific output."""
 
-    def get_gt_seg_maps(self, efficient_test=False):
+    def get_gt_seg_maps(self):
         """Get ground truth segmentation maps for evaluation."""
-        gt_seg_maps = []
         for img_info in self.img_infos:
             seg_map = osp.join(self.ann_dir, img_info['ann']['seg_map'])
-            if efficient_test:
-                gt_seg_map = seg_map
-            else:
-                gt_seg_map = mmcv.imread(
-                    seg_map, flag='unchanged', backend='pillow')
-            gt_seg_maps.append(gt_seg_map)
+            gt_seg_map = mmcv.imread(
+                seg_map, flag='unchanged', backend='pillow')
+            yield [gt_seg_map]
+
+    def index_gt_seg_maps(self, indexes):
+        """Get ground truth segmentation map by index for evaluation."""
+        if not isinstance(indexes, Iterable):
+            indexes = [indexes]
+        gt_seg_maps = []
+        for index in indexes:
+            seg_map = osp.join(self.ann_dir,
+                               self.img_infos[index]['ann']['seg_map'])
+            gt_seg_maps.append(
+                mmcv.imread(seg_map, flag='unchanged', backend='pillow'))
         return gt_seg_maps
 
```

Comment:
https://docs.python.org/3/library/itertools.html#itertools.islice

Reply:
Something like `islice(rank, len(dataset), world_size)` should handle samples on different gpus.